File: app/src/main/java/com/example/protect/ui/screens/DigitalWarrantyScreen.kt
package com.example.protect.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.protect.theme.*
import com.example.protect.ui.components.InfoBanner
import com.example.protect.ui.components.ProtectTopBar
import com.example.protect.ui.components.StatusBadge

@Composable
fun DigitalWarrantyScreen(
    onBack: () -> Unit,
    onClaimWarranty: () -> Unit
) {
    var isClaimed by remember { mutableStateOf(false) }

    Scaffold(
        topBar = {
            ProtectTopBar(
                title = "E-Sertifikat Garansi 12 Bulan",
                onBackClick = onBack
            )
        }
    ) { innerPadding ->
        LazyColumn(
            modifier = Modifier
                .fillMaxSize()
                .background(BackgroundLight)
                .padding(innerPadding),
            contentPadding = PaddingValues(16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            // Certificate Card
            item {
                Card(
                    shape = RoundedCornerShape(20.dp),
                    colors = CardDefaults.cardColors(containerColor = SurfaceLight),
                    elevation = CardDefaults.cardElevation(defaultElevation = 4.dp),
                    modifier = Modifier
                        .fillMaxWidth()
                        .border(2.dp, Emerald600, RoundedCornerShape(20.dp))
                ) {
                    Column(
                        modifier = Modifier.padding(20.dp),
                        horizontalAlignment = Alignment.CenterHorizontally
                    ) {
                        // Official Brand Logo & Gold Seal Header
                        Row(
                            modifier = Modifier.fillMaxWidth(),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            com.example.protect.ui.components.ProtectBrandLogo(height = 24.dp)

                            Box(
                                modifier = Modifier
                                    .size(44.dp)
                                    .clip(CircleShape)
                                    .background(Emerald50)
                                    .border(2.dp, WarningColor, CircleShape),
                                contentAlignment = Alignment.Center
                            ) {
                                Icon(
                                    imageVector = Icons.Default.Verified,
                                    contentDescription = null,
                                    tint = WarningColor,
                                    modifier = Modifier.size(24.dp)
                                )
                            }
                        }

                        Spacer(modifier = Modifier.height(12.dp))

                        Text(
                            text = "SERTIFIKAT GARANSI PERLINDUNGAN",
                            fontSize = 15.sp,
                            fontWeight = FontWeight.ExtraBold,
                            color = DarkSlate900,
                            letterSpacing = 1.sp,
                            textAlign = TextAlign.Center
                        )

                        Text(
                            text = "No. Dokumen: CERT-2026-WAR-8821",
                            fontSize = 11.sp,
                            fontFamily = FontFamily.Monospace,
                            color = TextSecondary
                        )

                        Spacer(modifier = Modifier.height(16.dp))
                        HorizontalDivider()
                        Spacer(modifier = Modifier.height(16.dp))

                        // Details Breakdown
                        WarrantyDetailRow(label = "Pemilik Sertifikat", value = "PT. Boga Lestari Prima")
                        WarrantyDetailRow(label = "Layanan", value = "Termite Control & Barrier Treatment")
                        WarrantyDetailRow(label = "Alamat Properti", value = "Jl. Senopati Raya No. 88, Jaksel")
                        WarrantyDetailRow(label = "Tanggal Pengerjaan", value = "02 September 2026")
                        WarrantyDetailRow(label = "Masa Garansi", value = "12 Bulan Penuh (Hingga 02 Sep 2027)")
                        WarrantyDetailRow(label = "Teknisi Bertanggung Jawab", value = "Doni Pratama (ID: TECH-09)")

                        Spacer(modifier = Modifier.height(16.dp))

                        // Barcode & QR Verification Box
                        Surface(
                            shape = RoundedCornerShape(12.dp),
                            color = SurfaceVariantLight,
                            modifier = Modifier.fillMaxWidth()
                        ) {
                            Column(
                                modifier = Modifier.padding(14.dp),
                                horizontalAlignment = Alignment.CenterHorizontally
                            ) {
                                Icon(
                                    imageVector = Icons.Default.QrCode,
                                    contentDescription = null,
                                    tint = DarkSlate900,
                                    modifier = Modifier.size(64.dp)
                                )
                                Spacer(modifier = Modifier.height(6.dp))
                                Text(
                                    text = "SHA-256: 8f4a9b2c...e12d (HACCP Verified)",
                                    fontSize = 10.sp,
                                    fontFamily = FontFamily.Monospace,
                                    color = TextSecondary
                                )
                            }
                        }
                    }
                }
            }

            // Guarantee Benefits
            item {
                Card(
                    shape = RoundedCornerShape(16.dp),
                    colors = CardDefaults.cardColors(containerColor = SurfaceLight),
                    modifier = Modifier.border(1.dp, BorderColor, RoundedCornerShape(16.dp))
                ) {
                    Column(modifier = Modifier.padding(16.dp), verticalArrangement = Arrangement.spacedBy(10.dp)) {
                        Text(
                            text = "Ketentuan & Hak Garansi",
                            fontSize = 14.sp,
                            fontWeight = FontWeight.Bold,
                            color = DarkSlate900
                        )

                        BenefitItem(text = "Re-treatment 100% GRATIS jika ada tanda aktivitas hama selama masa aktif.")
                        BenefitItem(text = "Respon teknisi siaga prioritas dalam waktu kurang dari 2 jam.")
                        BenefitItem(text = "Menggunakan bahan botanical non-repellent aman bagi anak dan staf.")
                    }
                }
            }

            // Claim Action Button
            item {
                if (isClaimed) {
                    InfoBanner(
                        text = "Klaim Garansi Telah Dikirim! Tim Teknisi Siaga PROTECT akan menghubungi Anda dalam 15 menit.",
                        icon = Icons.Default.CheckCircle,
                        backgroundColor = Emerald50,
                        contentColor = Emerald800
                    )
                } else {
                    Button(
                        onClick = { isClaimed = true },
                        colors = ButtonDefaults.buttonColors(containerColor = Emerald600),
                        shape = RoundedCornerShape(12.dp),
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(48.dp)
                            .testTag("btn_claim_warranty")
                    ) {
                        Icon(Icons.Default.SupportAgent, contentDescription = null, modifier = Modifier.size(18.dp))
                        Spacer(modifier = Modifier.width(8.dp))
                        Text("Ajukan Klaim Garansi / Re-Treatment", fontWeight = FontWeight.Bold)
                    }
                }
            }
        }
    }
}

@Composable
fun WarrantyDetailRow(label: String, value: String) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 4.dp),
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        Text(text = label, fontSize = 11.sp, color = TextSecondary)
        Text(
            text = value,
            fontSize = 11.sp,
            fontWeight = FontWeight.Bold,
            color = DarkSlate900,
            textAlign = TextAlign.End,
            modifier = Modifier.padding(start = 12.dp)
        )
    }
}

@Composable
fun BenefitItem(text: String) {
    Row(verticalAlignment = Alignment.Top) {
        Icon(
            imageVector = Icons.Default.CheckCircle,
            contentDescription = null,
            tint = Emerald600,
            modifier = Modifier.size(16.dp)
        )
        Spacer(modifier = Modifier.width(8.dp))
        Text(
            text = text,
            fontSize = 12.sp,
            color = DarkSlate900,
            lineHeight = 18.sp
        )
    }
}

File: lib/screens/digital_warranty_screen.dart
import 'package:flutter/material.dart';
import '../theme/app_theme.dart';
import '../widgets/common_widgets.dart';
import 'cs_chat_screen.dart';

class DigitalWarrantyScreen extends StatelessWidget {
  const DigitalWarrantyScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.backgroundLight,
      appBar: AppBar(
        title: const Text('Sertifikat E-Garansi Digital 12 Bulan'),
      ),
      bottomNavigationBar: Container(
        padding: const EdgeInsets.all(16),
        decoration: BoxDecoration(
          color: AppColors.surfaceLight,
          border: const Border(top: BorderSide(color: AppColors.borderColor)),
        ),
        child: SafeArea(
          child: ElevatedButton.icon(
            onPressed: () {
              Navigator.of(context).push(
                MaterialPageRoute(builder: (_) => const CsChatScreen()),
              );
            },
            icon: const Icon(Icons.verified_user, color: Colors.white),
            label: const Text('Klaim Re-Treatment Garansi', style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold)),
            style: ElevatedButton.styleFrom(
              backgroundColor: AppColors.emerald600,
              padding: const EdgeInsets.symmetric(vertical: 14),
              shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
            ),
          ),
        ),
      ),
      body: ListView(
        padding: const EdgeInsets.all(16),
        children: [
          // Certificate Card
          Container(
            padding: const EdgeInsets.all(20),
            decoration: BoxDecoration(
              color: AppColors.surfaceLight,
              borderRadius: BorderRadius.circular(20),
              border: Border.all(color: const Color(0xFFD97706), width: 2),
              boxShadow: [
                BoxShadow(
                  color: const Color(0xFFD97706).withOpacity(0.12),
                  blurRadius: 16,
                  offset: const Offset(0, 6),
                ),
              ],
            ),
            child: Column(
              children: [
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: [
                    Row(
                      children: [
                        Container(
                          padding: const EdgeInsets.all(6),
                          decoration: BoxDecoration(
                            color: AppColors.emerald600,
                            borderRadius: BorderRadius.circular(8),
                          ),
                          child: const Icon(Icons.shield, color: Colors.white, size: 20),
                        ),
                        const SizedBox(width: 8),
                        const Text(
                          'PROTECT PCO',
                          style: TextStyle(fontSize: 14, fontWeight: FontWeight.w900, color: AppColors.darkSlate900),
                        ),
                      ],
                    ),
                    const StatusBadge(
                      text: 'GOLD WARRANTY',
                      backgroundColor: Color(0xFFFEF3C7),
                      textColor: Color(0xFFB45309),
                    ),
                  ],
                ),
                const SizedBox(height: 16),
                const Text(
                  'SERTIFIKAT JAMINAN PROTEKSI RESMI',
                  style: TextStyle(
                    fontSize: 14,
                    fontWeight: FontWeight.w900,
                    letterSpacing: 1.2,
                    color: Color(0xFF92400E),
                  ),
                  textAlign: TextAlign.center,
                ),
                const SizedBox(height: 6),
                const Text(
                  'No. Sertifikat: PRT-GAR-2026-9921',
                  style: TextStyle(fontSize: 11, fontFamily: 'monospace', fontWeight: FontWeight.bold, color: AppColors.textSecondary),
                ),
                const Divider(height: 24),

                _buildCertRow('Penerima Sertifikat', 'PT. Boga Lestari Prima (Senopati Hub)'),
                const SizedBox(height: 8),
                _buildCertRow('Jenis Layanan', 'Termite Barrier & Rodent Control'),
                const SizedBox(height: 8),
                _buildCertRow('Tanggal Perlakuan', '02 September 2026'),
                const SizedBox(height: 8),
                _buildCertRow('Masa Berlaku Garansi', '02 Sep 2026 s/d 02 Sep 2027 (12 Bulan)'),
                const SizedBox(height: 8),
                _buildCertRow('Teknisi Penanggung Jawab', 'Doni Pratama (ID: TECH-09)'),
                const Divider(height: 24),

                Container(
                  padding: const EdgeInsets.all(12),
                  decoration: BoxDecoration(
                    color: AppColors.backgroundLight,
                    borderRadius: BorderRadius.circular(10),
                  ),
                  child: Row(
                    children: [
                      const Icon(Icons.qr_code_2, size: 48, color: AppColors.darkSlate900),
                      const SizedBox(width: 12),
                      Expanded(
                        child: Column(
                          crossAxisAlignment: CrossAxisAlignment.start,
                          children: const [
                            Text(
                              'SHA-256 Digital Verification Hash:',
                              style: TextStyle(fontSize: 9, color: AppColors.textSecondary),
                            ),
                            Text(
                              'a9f82c3...7e1b4028 (Verified Active)',
                              style: TextStyle(fontSize: 10, fontFamily: 'monospace', fontWeight: FontWeight.bold, color: AppColors.emerald700),
                            ),
                          ],
                        ),
                      ),
                    ],
                  ),
                ),
              ],
            ),
          ),
          const SizedBox(height: 16),

          // Warranty Terms
          const SectionHeader(title: 'Ketentuan Jaminan Re-Treatment'),
          const SizedBox(height: 8),
          Container(
            padding: const EdgeInsets.all(16),
            decoration: BoxDecoration(
              color: AppColors.surfaceLight,
              borderRadius: BorderRadius.circular(16),
              border: Border.all(color: AppColors.borderColor),
            ),
            child: Column(
              children: const [
                _WarrantyTermItem(
                  number: '1',
                  text: 'Jika ditemukan kembali aktivitas hama target selama masa garansi aktif, teknisi akan melakukan re-treatment gratis tanpa biaya tambahan.',
                ),
                SizedBox(height: 10),
                _WarrantyTermItem(
                  number: '2',
                  text: 'Pengendalian menggunakan formula ramah lingkungan berstandar HACCP dan aman bagi area pengolahan makanan.',
                ),
                SizedBox(height: 10),
                _WarrantyTermItem(
                  number: '3',
                  text: 'Klaim dapat diajukan secara instan melalui aplikasi atau hotline CS 24/7 dengan waktu respon maksimal 4 jam kerja.',
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildCertRow(String label, String value) {
    return Row(
      mainAxisAlignment: MainAxisAlignment.spaceBetween,
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(label, style: const TextStyle(fontSize: 11, color: AppColors.textSecondary)),
        const SizedBox(width: 8),
        Expanded(
          child: Text(
            value,
            style: const TextStyle(fontSize: 11, fontWeight: FontWeight.bold, color: AppColors.darkSlate900),
            textAlign: TextAlign.end,
          ),
        ),
      ],
    );
  }
}

class _WarrantyTermItem extends StatelessWidget {
  final String number;
  final String text;

  const _WarrantyTermItem({required this.number, required this.text});

  @override
  Widget build(BuildContext context) {
    return Row(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Container(
          width: 18,
          height: 18,
          decoration: const BoxDecoration(
            color: AppColors.emerald50,
            shape: BoxShape.circle,
          ),
          child: Center(
            child: Text(
              number,
              style: const TextStyle(fontSize: 10, fontWeight: FontWeight.bold, color: AppColors.emerald700),
            ),
          ),
        ),
        const SizedBox(width: 8),
        Expanded(
          child: Text(text, style: const TextStyle(fontSize: 12, color: AppColors.textSecondary, height: 1.4)),
        ),
      ],
    );
  }
}
